### PR TITLE
Improve responsive mode for mobile devices #3

### LINKED ISSUE #3: Improve spacing on mobile landscape mode
When mobile is flipped to landscape mode the links-bar looks kind of crowded. I think we could improve the proportions (possibly decrease the margin between the header and the quote)

image

### PR DESCRIPTION
**Pull request by mrbjnb**
- Add reset.css for reseting the format of all browser.
- Improve responsive mode on mobile devices by adjusting the margin and font-size of header and quote.
- Make the same changes with smaller size.

### CODE REVIEW

File: site/reset.css
Comment (aredshift) [inline site/reset.css:1]: Interesting! I'm learning about CSS resets and normalization for the first time right now. Why did you decide to add this reset vs something like [modern-normalize](https://github.com/sindresorhus/modern-normalize) or [sanitize.css](https://github.com/csstools/sanitize.css)?

File: site/index.css
Comment (aredshift) [inline site/index.css]: I'd like this size to stay at its default 1em.

File: site/index.css
Comment (aredshift) [inline site/index.css]: These changes end up making the text too large on portrait-mode mobile screens.

Additionally, the change to `#aquapants-header p` is redundant (and like I mentioned, I think it looks too big and should be 1em)

### Samsung Galaxy
mrbjnb-pr-galaxy

File: site/reset.css
Comment (aredshift) [inline site/reset.css]: This removes the quotation marks, which I think should be added back.

File: site/index.css
Comment (aredshift) [inline site/index.css]: I think this is a good implementation, but I don't like how it looks on taller screens (too crowded). Can we use a media query to only apply this rule to mobile horizontal screens (ie. max-height is 620px)?

**Review by aredshift — CHANGES_REQUESTED** [review]

**Comment by aredshift** [comment]
Also, FYI, it is good practice to describe what changes you made and why you made them in the Pull Request description. This helps reviewers read your code and also helps future contributors come back to your PR and understand why we did things the way we did.

https://www.freecodecamp.org/news/how-to-write-a-pull-request-description/
https://www.pullrequest.com/blog/writing-a-great-pull-request-description/

No need to go into as much detail as the articles say, but some commentary would help a lot.

This is also partially our fault because we haven't provided a template yet. I'll make sure we do that!

**Comment by aredshift** [comment]
Also, when opening a PR that is addressing an existing issue in Github, it's particularly useful to mark the pull request using a simple comment like "closes  #3" or alternatively, use the UI. More instructions here:

1. [Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
2. [Using keywords in issues and pull requests](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

File: site/index.css
Comment (aredshift) [reply site/index.css]: Just realized that [I asked you to make this bigger](https://github.com/aquapants/website/issues/3#issuecomment-1792950377) :sweat_smile: 

What resolution are you using to test this? I'm guessing the [header and subheader are getting too small](https://github.com/aquapants/website/issues/3#issuecomment-1792848159) because your width is less than 620px (thus the media query thinks it's a mobile phone in portrait orientation). If that's the case, try testing with some common device dimensions instead. Firefox's [responsive design mode](https://firefox-source-docs.mozilla.org/devtools-user/responsive_design_mode/index.html) has good options for this, for example

File: site/reset.css
Comment (mrbjnb) [reply site/reset.css:1]: Because I couldn't find a way to reduce the empty space between header and p tag, I found a post said that you should include this in your webpage to reset the default setting in every browser.
Reference: https://stackoverflow.com/questions/14423781/how-can-i-remove-space-margin-around-an-html-header

File: site/index.css
Comment (mrbjnb) [reply site/index.css]: I use Chrome developer tool and adjust the width to 620px and less, maybe some device has bigger width so it doesn't work with them, I can try to figure the general width size.

File: site/index.css
Comment (mrbjnb) [reply site/index.css]: Yes I can do that, I thought how much I can put in that max-width:620px query 😅 cause that that what the requirement is for (landscape mode)

File: site/reset.css
Comment (mrbjnb) [reply site/reset.css]: I will add it back to the index.css. Because this is the default reset.css for reseting every browser to 0 so I rather not touching it.

File: site/reset.css
Comment (mrbjnb) [reply site/reset.css:1]: I am new to this too so I just pick the one that I found, I will test two of the normalize css that you mentioned. However I think the function is still the same, which is reseting any browser default format.

**Comment by mrbjnb** [comment]
Lmk if I need to add anything more.

File: site/reset.css
Comment (aredshift) [reply site/reset.css]: Good work! I'll just remark here that reading Eric Meyer's website, he did intend for the reset.css script to be tweaked as needed! So going forward we should feel free to tweak about as we please.

> I don't particularly recommend that you just use this in its unaltered state in your own projects. It should be tweaked, edited, extended, and otherwise tuned to match your specific reset baseline. Fill in your preferred colors for the page, links, and so on. 

https://meyerweb.com/eric/tools/css/reset/

File: site/reset.css
Comment (aredshift) [reply site/reset.css:1]: Neat! We can create a separate issue to mess around with the reset logic, I suppose. I support the idea of adding a reset in any case, though, and we can always replace it with a more modern one like the two above if need be. Here's a fun, recently published read regarding CSS resets: https://mattbrictson.com/blog/css-normalize-and-reset

File: site/index.css
Comment (aredshift) [reply site/index.css]: Ah I see, maybe I should have commented that block. When I added the `max-width:620px` query, I meant for that to be portrait mode. The idea of using 620px as a breakpoint was loosely based on [this article](https://www.freecodecamp.org/news/the-100-correct-way-to-do-css-breakpoints-88d6a5ba1862/) and some arguments I read on StackOverflow regarding common breakpoints.

I think since landscape mode is just rotated portrait-mode, we should simply treat it with a `max-height:620px` media query. Your current implementation does not apply the style on most landscape-orientation phones, which often have a width (ie. the height of the phone in portrait mode) that is larger than 620px.

**Comment by aredshift** [comment]
@mrbjnb Great improvements! Just a few more suggestions for cleaning it up before we merge and deploy.

Also note, on iPhone 12/13 portrait-mode (375/812) the quote looks a bit squished on the sides. We could probably solve this by adding a margin on both sides (I'm guessing the added css reset removed whatever margins used to be there).

Screenshot:
image

File: site/reset.css
Comment (mrbjnb) [reply site/reset.css]: Oh I see, then hmm, I should modify the reset.css then.

**Comment by mrbjnb** [comment]
Oh I tried to change the name of the branch, I guess it doesn't work xd

File: site/index.css
Comment (aredshift) [inline site/index.css:72]: This is why breakpoints are tricky! Here's an iPhone version that has a width of 428 px and therefore it falls under the "tablet" category. I think to be safe let's just use `620px` as our breakpoint for mobile L width.

```suggestion
@media screen and (max-width: 620px) {
```

image

File: site/index.css
Comment (aredshift) [inline site/index.css]: We can remove this since we're adding the margin to all sizes above.

```suggestion
```

File: site/index.css
Comment (aredshift) [inline site/index.css]: We can also remove this since we're adding the margin to all sizes above.

```suggestion
```

File: site/index.css
Comment (aredshift) [inline site/index.css]: Let's just add the margin to all sides of the quote block so that it looks symmetrical.

```suggestion
  margin: 1.5em;
```

image

**Review by aredshift — CHANGES_REQUESTED** [review]
Thank you @mrbjnb! I have just a couple more nitpicks before we close this one out.

File: site/index.css
Comment (mrbjnb) [reply site/index.css:72]: Ah, I found the problem, I was making changes to 768px, 425px and 320px mode. I accidentally put the "max-width" to "max-height", which cause the above problem.

Comment (mrbjnb) [reply site/index.css:72]: https://github.com/aquapants/website/pull/9/commits/876f33eb43fd54cd2969b41f1f86437157e55a68 
Fixed

**Review by aredshift — APPROVED** [review]
Sorry for the delay - Looks great to me! Thanks @mrbjnb for all your work, well done! We'll merge this and I'll notify you when the changes are deployed.

**Comment by aredshift** [comment]
@mrbjnb your changes are officially deployed to [aquapants.org](https://aquapants.org)!

**Pull request merged**
